scripts/install

```diff
@@ -1,189 +1,208 @@
-ynh_script_progression --message="Creating database..." --weight=1
+ynh_script_progression --message="Creating a PostgreSQL database..." --weight=1
+
 db_name=$(ynh_sanitize_dbid $app)
+db_user=$db_name
 ynh_app_setting_set --app=$app --key=db_name --value=$db_name
 ynh_psql_test_if_first_run
-ynh_exec_warn_less ynh_psql_setup_db --db_user=$db_name --db_name=$db_name
-db_pass=$(ynh_app_setting_get --app=$app --key=psqlpwd)
+ynh_psql_setup_db --db_user=$db_user --db_name=$db_name
+db_pwd=$(ynh_app_setting_get --app=$app --key=psqlpwd)
 
 #=================================================
-# EXPORT VARIABLES FOR TEMPLATING
+# DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
-export app
-export domain
-export db_pass
-export final_path
-export admin
+ynh_script_progression --message="Setting up source files..." --weight=1
 
-#=================================================
-# CONFIGURE DIASPORA
-#=================================================
-ynh_script_progression --message="Configure diaspora..." --weight=1
-ynh_render_template ../conf/diaspora.yml $final_path/diaspora/config/diaspora.yml
-ynh_render_template ../conf/database.yml $final_path/diaspora/config/database.yml
+ynh_app_setting_set --app=$app --key=final_path --value=$final_path
+# Download, check integrity, uncompress and patch the source from app.src
+ynh_setup_source --dest_dir="$final_path/live"
+
+chmod 750 "$final_path"
+chmod -R o-rwx "$final_path"
+chown -R $app:www-data "$final_path"
 
 #=================================================
-# STORE THE CHECKSUM OF THE CONFIG FILE
+# NGINX CONFIGURATION
 #=================================================
-# Calculate and store the config file checksum into the app settings
-ynh_store_file_checksum --file="$final_path/diaspora/config/diaspora.yml"
-ynh_store_file_checksum --file="$final_path/diaspora/config/database.yml"
+ynh_script_progression --message="Configuring NGINX web server..." --weight=1
+
+# Create a dedicated NGINX config
+ynh_add_nginx_config
 
 #=================================================
-# Bundle the ruby app
+# SPECIFIC SETUP
+#=================================================
+# ADD A CONFIGURATION
 #=================================================
-ynh_script_progression --message="Precompile assets (will take a long time)..." --weight=40
-source ./bundle_app
+ynh_script_progression --message="Adding a configuration file..." --weight=1
+
+ynh_add_config --template="../conf/diaspora.yml" --destination="$final_path/live/config/diaspora.yml"
+chmod 400 "$final_path/live/config/diaspora.yml"
+chown $app:$app "$final_path/live/config/diaspora.yml"
+
+ynh_add_config --template="../conf/database.yml" --destination="$final_path/live/config/database.yml"
+chmod 400 "$final_path/live/config/database.yml"
+chown $app:$app "$final_path/live/config/database.yml"
 
 #=================================================
-# NGINX CONFIGURATION
+# BUILD APP
```

Comment:
In the same spirit as the templating engine remark, I with the standard way of packaging apps didn't involve so many code duplications... That's why I introduced `bundle_app` and `create_service` that this PR removes...